File: Python/Ejercicios-LOO/ejercicio-18.py
# 18. Implementar una función que recibe dos enteros de 8 dígitos con el formato aaaammdd, informar cuál de las dos es la anterior y cuál la posterior. Usar lo aplicado en ejercicio anterior

fechaUno = int(input("Ingrese fecha uno: "))
fechaDos = int(input("Ingrese fecha dos: "))

# Comparar años
if round(fechaUno // 10000) > round(fechaDos // 10000):
    print(fechaUno, "es mayor que", fechaDos)
else:
    if round(fechaUno // 10000) == round(fechaDos // 10000):
        # Comparar meses
        if round((fechaUno / 100) % 100) > round((fechaDos / 100) % 100):
            print(fechaUno, "es mayor que", fechaDos)
        else:
            if round((fechaUno / 100) % 100) == round((fechaDos / 100) % 100):
                # Comparar días
                if round(fechaUno % 100) > round(fechaDos % 100):
                    print(fechaUno, "es mayor que", fechaDos)
                else:
                    print(fechaDos, "es mayor que", fechaUno)
        print(fechaDos, "es mayor que", fechaUno)
    else:
        print(fechaDos, "es mayor que", fechaUno)   

File: Java/src/EjerciciosLOO/Ejercicio_18.java
package EjerciciosLOO;

import java.util.Scanner;

public class Ejercicio_18 {
    public static void main(String[] args) {
        // Implementar una función que recibe dos enteros de 8 dígitos con el formato
        // aaaammdd, informar cuál de las dos es la anterior y cuál la posterior. Usar
        // lo aplicado en ejercicio anterior. 

        Scanner sc = new Scanner(System.in);

        System.out.println("Ingrese fecha uno:");
        int fechaUno = sc.nextInt();
        System.out.println("Ingrese fecha dos:");
        int fechaDos = sc.nextInt();

        sc.close();

        if (Math.round(fechaUno / 10000) > Math.round(fechaDos / 10000)) {
            System.out.println(fechaUno + " es mayor que " + fechaDos);
        } else {
            if (Math.round(fechaUno / 10000) == Math.round(fechaDos / 10000)) {
                if (Math.round(fechaUno / 100) % 100 > Math.round(fechaDos / 100) % 100) {
                    System.out.println(fechaUno + " es mayor que " + fechaDos);
                } else {
                    if (Math.round(fechaUno / 100) % 100 == Math.round(fechaDos / 100) % 100) {
                        if (Math.round(fechaUno % 100) > Math.round(fechaDos % 100)) {
                            System.out.println(fechaUno + " es mayor que " + fechaDos);
                        } else {
                            System.out.println(fechaDos + " es mayor que " + fechaUno);
                        }
                    } else {
                        System.out.println(fechaDos + " es mayor que " + fechaUno);
                    }
                }
            } else {
                System.out.println(fechaDos + " es mayor que " + fechaUno);
            }
        }
    }
}
